CMakeLists.txt

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+# Copyright (c) 2020 - 2022 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.16)
+
+# check if iceoryx is in CMAKE_PREFIX_PATH
+find_package(iceoryx_hoofs QUIET)
+find_package(iceoryx_dust QUIET)
+find_package(iceoryx_posh QUIET)
+
+# Link iceoryx_hoofs and iceoryx_posh libraries statically
+option(BUILD_SHARED_LIBS "Link libraries dynamically" OFF)
+
+# fetch iceoryx if not found
+if(NOT iceoryx_hoofs_FOUND OR NOT iceoryx_posh_FOUND OR NOT iceoryx_dust_FOUND)
+    if(iceoryx_hoofs_FOUND)
+        message(FATAL_ERROR "iceoryx_hoofs was not found with 'find_package' but other parts were found!")
+    endif()
+    if(iceoryx_dust_FOUND)
+        message(FATAL_ERROR "iceoryx_dust was not found with 'find_package' but other parts were found!")
+    endif()
+    if(iceoryx_posh_FOUND)
+        message(FATAL_ERROR "iceoryx_posh was not found with 'find_package' but other parts were found!")
+    endif()
+
+    include(FetchContent)
+    FetchContent_Declare(
+        iceoryx
+        GIT_REPOSITORY https://github.com/eclipse-iceoryx/iceoryx.git
+        # Recent commit used here to be able to use the CMake macros. Change this to v3.x once released.
+        GIT_TAG be2272fee0b25ce203fa63cc3a93a740baaa4c73
```

Comment:
This should be changed the merge commit of the removal of the gateway from the main repo

Reply:
Fixed.